### DISCUSSION TITLE
blog/specification-pattern-in-aspnet-core/

### DISCUSSION THREAD

**Discussion by giscus** [codewithmukesh/discussions#24] [General]
# blog/specification-pattern-in-aspnet-core/

We will talk about implementing Specification Pattern in ASP.NET Core applications and how it can enhance the already existing Generic Repository Patterns.

https://codewithmukesh.com/blog/specification-pattern-in-aspnet-core/

**Comment by alexandra-valkova** [comment]
Thanks for this article, I found it very useful!

I noticed though that the `SpecificationEvaluator<T>` class implementation is missing here, but I managed to find it in the [github repository](https://github.com/iammukeshm/specification-pattern-asp-net-core/blob/3bf62b3cef425835a011ba21d42933c46a56f992/Data/SpecificationEvaluator.cs).

Also, for some reason the syntax highlighting doesn't work on this post.

**Comment by anurag0431** [comment]
Nice article.
But Can you please explain how we can Include enities, If we have multiple Include and then ThenInclude .

something like this
context.Blogs
        .Include(blog => blog.Posts)
            .ThenInclude(post => post.Author)
        .Include(blog => blog.Posts)
            .ThenInclude(post => post.Tags)
